### DISCUSSION TITLE
How can I upload big files on github<title>

### DISCUSSION THREAD

**Discussion by ganesha2208** [community/community#49949] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

I want to upload big file on github how can I

**Comment by neutrinomuon** [comment, marked as answer]
Hi,

GitHub has a file size limit for individual files that can vary according to the plan. If you need to upload a file that is larger than this, you have a few options:

1. Use Git Large File Storage (LFS): I like this option. Git LFS (https://git-lfs.com/) is an extension of Git that allows you to store large files outside of your repository, while still keeping track of their versions. You can install Git LFS on your local machine and on the GitHub server, and then use the "git lfs" command to manage large files in your repository.

2. Use a third-party storage service: If your file is too large to be stored on GitHub, you can consider using a third-party storage service such as Amazon S3 or Google Cloud Storage. You can upload your file to the storage service, and then add a link to the file in your GitHub repository.

3. Not ideal, but you can split the file into smaller parts: If possible, you can split your file into smaller parts that are each under XX Mb in size. You can then upload each part to your GitHub repository and use a tool like "cat" to combine the parts into the original file when you need to use it.

I hope this helps! Let me know if you have any further questions. ʕ•́ᴥ•̀ʔっ

**Reply by shavez12343-art** [reply]
big

**Comment by Anmol-Baranwal** [comment]
By default, GitHub has a file size limit of 25 MB for each file that you upload. However, there are several ways to upload larger files to GitHub:

1. Git Large File Storage (Git LFS): Git LFS is an extension to Git that allows you to store large files separately from your Git repository. GitHub supports Git LFS, and you can use it to store files larger than 25 MB. To use Git LFS, you'll need to install it on your local machine and configure it for your repository. You can find more information about Git LFS on the GitHub website.

2. GitHub Desktop: GitHub Desktop is a graphical user interface for Git that makes it easier to work with Git repositories. GitHub Desktop has built-in support for Git LFS, so you can use it to upload large files to GitHub. Simply drag and drop the files into the repository, and GitHub Desktop will take care of the rest.

3. Command Line: If you prefer to use the command line, you can use Git LFS directly to upload large files to GitHub. First, you'll need to install Git LFS on your local machine and configure it for your repository. Then, use the `git lfs` command to add the large files to Git LFS and push them to GitHub.

It's important to note that uploading large files to GitHub can affect the performance of your repository, especially if you have a slow internet connection. It's a good idea to use Git LFS or other methods to manage large files separately from your Git repository to keep your repository lean and fast.

**Reply by Anmol-Baranwal** [reply]
@ganesha2208 
I just wanted to check in and see if you had any further questions or concerns about my answer on GitHub. If you have no further doubts or if your doubts have been resolved, would you mind marking the most helpful answer as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by MNLierman** [reply]
> @ganesha2208 I just wanted to check in and see if you had any further questions or concerns about my answer on GitHub. If you have no further doubts or if your doubts have been resolved, would you mind marking the most helpful answer as accepted?
> 
> It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

I have further questions. Unless I am misunderstanding something, you are indicating that GitHub has a max file size limit of 25MB, and if one needs larger file sizes, they would need to upload files outside the repo. However, throughout my experience with GitHub, I have seen many repos that upload large files, ISOs, large release files, some in a zip some not in a zip. Some files are GBs in size. How are these repos uploading such large files, if you say that larger than 25MB would have to be uploaded outside the repo using LFS? I'm not quite understanding.

**Reply by Sky-Way2** [reply]
@MNLierman these are bigger github users that work with Github to make their file size limits more than average people.
